radp/digital_twin/mobility/ue_generation_handler.py

```diff
@@ -0,0 +1,257 @@
+            max_lat = self.max_lat,
+            min_lon = self.min_lon,
+            max_lon = self.max_lon
+            ) 
+
+        all_batches = []
+        for batch_no in range(self.num_batches):
+            batch = next(ue_generator.generate())
+            all_batches.append(batch)
+        print("Batches of Data: " , all_batches)
+        return all_batches
+
+
+
+    def mobility_data_generation(self) -> pd.DataFrame:
+
+        """
+        Generates UE Mobility Data, converts X and Y Co-ordinates to Latitude and Longitude respectively,
+        and organises everything into dataframes of the following format:
+            +------------+------------+-----------+------+
+            | mock_ue_id | lon        | lat       | tick |
+            +============+============+===========+======+
+            |   0        | 102.219377 | 33.674572 |   0  |
+            |   1        | 102.415954 | 33.855534 |   0  |
+            |   2        | 102.545935 | 33.878075 |   0  |
+            |   0        | 102.297766 | 33.575942 |   1  |
+            |   1        | 102.362725 | 33.916477 |   1  |
+            |   2        | 102.080675 | 33.832793 |   1  |
+            +------------+------------+-----------+------+
+            
+        return: A pandas DataFrame containing the UE tracks data of all the batches.
+        """
+        
+        ue_data = self.generate_ue_data()
+        all_dataframes = []  # List to hold all batch dataframes
+
+        # Iterate through each batch
+        for batch_no, xy_batches in enumerate(ue_data):
+            batch_dataframe_list = []
+
+          # Iterate through ticks in each batch
+            for tick, xy_batch in enumerate(xy_batches):
+              # Transforming (x, y) into (lon, lat)
+                lon_lat_pairs = GISTools.converting_xy_points_into_lonlat_pairs(
+                  xy_batch,
+                  self.lon_x_dims,
+                  self.lon_y_dims,
+                  self.min_lon,
+                  self.max_lon,
+                  self.min_lat,
+                  self.max_lat
+                )
+
+                # Building DataFrame for this tick
+                df_tick = pd.DataFrame({
+                    'mock_ue_id': range(self.num_UEs),
+                    'longitude': [pair[0] for pair in lon_lat_pairs],
+                    'latitude': [pair[1] for pair in lon_lat_pairs],
+                    'tick': np.full(self.num_UEs, tick)
+                })
+
+                batch_dataframe_list.append(df_tick)
+
+            # Concatenate all ticks data for the current batch
+            all_dataframes.append(pd.concat(batch_dataframe_list, ignore_index=True))
+
+            if batch_no + 1 >= self.num_batches:  # Check if the specified number of batches has been reached
+                break
+
+        # Concatenate all batches into a single DataFrame and return
+        return pd.concat(all_dataframes, ignore_index=True)
+
+
+    def plot_ue_tracks(self,csv_file) -> None:
```

Comment:
make this a static method inside ue_tracks that takes a UETracksGenerator object and simulates and plots it. The whole simulation can be done and the whole data can be loaded into memory and plotted (as is being done here) OR, if possible, just enumerate through generate_as_lon_lat_points and plot one by one.